src/common/screen.rs

```diff
@@ -330,3 +314,188 @@ impl Screen {
         }
     }
 }
+
+pub fn screen_thread_main(
+    bus: Bus<ScreenInstruction>,
+    mut command_is_executing: CommandIsExecuting,
+    max_panes: Option<usize>,
+    full_screen_ws: PositionAndSize,
+) {
+    let mut screen = Screen::new(bus, &full_screen_ws, max_panes, ModeInfo::default());
+    loop {
+        let (event, mut err_ctx) = screen
+            .bus
+            .recv()
+            .expect("failed to receive event on channel");
+        err_ctx.add_call(ContextType::Screen(ScreenContext::from(&event)));
+        match event {
```

Comment:
It seems that here we have the match in the `thread_main` function, but the `Pty` has a separate function for handling the events (the `match`) and the main loop (`thread_main` function). I'm somewhat partial to the having everything together in the same `thread_main` like this and the WASM thread, that seems quite clean, but I don't really mind either so long as we're consistent :)

Reply:
Replied to this below.